compile error on macOS

On macOS with Apple clang version 14.0.0 I get the error

```
Consolidate compiler generated dependencies of target umfpack_int
[ 31%] Building C object CMakeFiles/umfpack_int.dir/UMFPACK/Source/umf_cholmod.c.o
/Users/fwein/Downloads/SuiteSparse-cmake/UMFPACK/Source/umf_cholmod.c:175:9: error: implicit declaration of function 'cholmod_analyze' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
    L = CHOLMOD_analyze (S, &cm) ;
        ^
/Users/fwein/Downloads/SuiteSparse-cmake/UMFPACK/Source/umf_cholmod.c:24:29: note: expanded from macro 'CHOLMOD_analyze'
#define CHOLMOD_analyze     cholmod_analyze
                            ^
/Users/fwein/Downloads/SuiteSparse-cmake/UMFPACK/Source/umf_cholmod.c:175:9: note: did you mean 'cholmod_scale'?
/Users/fwein/Downloads/SuiteSparse-cmake/UMFPACK/Source/umf_cholmod.c:24:29: note: expanded from macro 'CHOLMOD_analyze'
#define CHOLMOD_analyze     cholmod_analyze
                            ^
/Users/fwein/Downloads/SuiteSparse-cmake/CHOLMOD/Include/cholmod_matrixops.h:122:5: note: 'cholmod_scale' declared here
int cholmod_scale
    ^
/Users/fwein/Downloads/SuiteSparse-cmake/UMFPACK/Source/umf_cholmod.c:175:7: warning: incompatible integer to pointer conversion assigning to 'cholmod_factor *' (aka 'struct cholmod_factor_struct *') from 'int' [-Wint-conversion]
    L = CHOLMOD_analyze (S, &cm) ;
```
The `-Werror` puzzles me, as it seems it could be avoided, however it is unclear how to avoid it. The compile call is

```
/Library/Developer/CommandLineTools/usr/bin/cc -DBLAS_UNDERSCORE -DDINT -Dumfpack_EXPORTS -I/Users/fwein/Downloads/SuiteSparse-cmake/AMD/Include -I/Users/fwein/Downloads/SuiteSparse-cmake/UMFPACK/Include -I/Users/fwein/Downloads/SuiteSparse-cmake/UMFPACK/Source -I/Users/fwein/Downloads/SuiteSparse-cmake/build/include/SuiteSparse/UMFPACK -I/Users/fwein/Downloads/SuiteSparse-cmake/build/CHOLMOD/Include -I/Users/fwein/Downloads/SuiteSparse-cmake/CHOLMOD/Cholesky -I/Users/fwein/Downloads/SuiteSparse-cmake/CHOLMOD/Core -I/Users/fwein/Downloads/SuiteSparse-cmake/CHOLMOD/Include -I/Users/fwein/Downloads/SuiteSparse-cmake/CHOLMOD/MatrixOps -I/Users/fwein/Downloads/SuiteSparse-cmake/CHOLMOD/Modify -I/Users/fwein/Downloads/SuiteSparse-cmake/CHOLMOD/Supernodal -I/Users/fwein/Downloads/SuiteSparse-cmake/build/include/SuiteSparse/CHOLMOD -I/Users/fwein/Downloads/SuiteSparse-cmake/SuiteSparse_config -I/Users/fwein/Downloads/SuiteSparse-cmake/build/include/SuiteSparse -I/Users/fwein/Downloads/SuiteSparse-cmake/build/include/SuiteSparse/AMD -O3 -DNDEBUG -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk -mmacosx-version-min=12.3 -fPIC -fvisibility=hidden -MD -MT CMakeFiles/umfpack_int.dir/UMFPACK/Source/umf_cholmod.c.o -MF CMakeFiles/umfpack_int.dir/UMFPACK/Source/umf_cholmod.c.o.d -o CMakeFiles/umfpack_int.dir/UMFPACK/Source/umf_cholmod.c.o -c /Users/fwein/Downloads/SuiteSparse-cmake/UMFPACK/Source/umf_cholmod.c
```

I don't know if this can be avoided by the project or if is an issue of the original sparse suite. The current sparse suite 6.0 beta compiles easily.

BTW, all on but only `WITH_CHOLESKY`
does also not work

```
[ 19%] Building CXX object CMakeFiles/spqr.dir/SPQR/Source/spqr_1colamd.cpp.o
/Users/fwein/Downloads/SuiteSparse-cmake/SPQR/Source/spqr_1colamd.cpp:564:13: error: use of undeclared identifier 'cholmod_l_amd'
            cholmod_l_amd (AT, NULL, 0, (Long *) (Q1fill + n1cols), cc) ;
            ^
/Users/fwein/Downloads/SuiteSparse-cmake/SPQR/Source/spqr_1colamd.cpp:581:18: error: use of undeclared identifier 'cholmod_l_analyze_p2'
            Sc = cholmod_l_analyze_p2 (FALSE, AT, NULL, NULL, 0, cc) ;
                 ^
/Users/fwein/Downloads/SuiteSparse-cmake/SPQR/Source/spqr_1colamd.cpp:606:13: error: use of undeclared identifier 'cholmod_l_colamd'; did you mean 'cholmod_l_ccolamd'?
            cholmod_l_colamd (AT, NULL, 0, TRUE,
            ^~~~~~~~~~~~~~~~
            cholmod_l_ccolamd
/Users/fwein/Downloads/SuiteSparse-cmake/CHOLMOD/Include/cholmod_camd.h:61:5: note: 'cholmod_l_ccolamd' declared here
int cholmod_l_ccolamd (cholmod_sparse *, SuiteSparse_long *, size_t,
    ^
/Users/fwein/Downloads/SuiteSparse-cmake/SPQR/Source/spqr_1colamd.cpp:606:44: error: cannot initialize a parameter of type 'long *' with an rvalue of type 'int'
            cholmod_l_colamd (AT, NULL, 0, TRUE,
                                           ^~~~
/Users/fwein/Downloads/SuiteSparse-cmake/SPQR/Include/spqr.hpp:67:14: note: expanded from macro 'TRUE'
#define TRUE 1
             ^
/Users/fwein/Downloads/SuiteSparse-cmake/CHOLMOD/Include/cholmod_camd.h:62:23: note: passing argument to parameter here
    SuiteSparse_long *, SuiteSparse_long *, cholmod_common *) ;
                      ^
4 errors generated.

```

But again, this is on issue, just an observation.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -87,7 +87,8 @@ cmake_dependent_option (WITH_CAMD "Enable interfaces to CAMD, CCOLAMD, CSYMAMD i
 cmake_dependent_option (WITH_CHECK "Enable the Check module" ON "WITH_LGPL" OFF)
 cmake_dependent_option (WITH_MATRIXOPS "Enable the MatrixOps module" ON "WITH_GPL AND WITH_CHOLMOD" OFF)
 cmake_dependent_option (WITH_MODIFY "Enable the Modify module" ON "WITH_GPL AND WITH_CHOLMOD" OFF)
-cmake_dependent_option (WITH_SUPERNODAL "Enable the Supernodal module" ON "CMAKE_CXX_COMPILER AND WITH_GPL" OFF)
+cmake_dependent_option (WITH_SUPERNODAL "Enable the Supernodal module" ON
+  "WITH_CHOLESKY AND CMAKE_CXX_COMPILER AND WITH_GPL" OFF)
 
 if (WITH_CUDA)
   enable_language (CUDA)
@@ -1591,7 +1592,7 @@ if (WITH_DEMOS)
   )
   target_link_libraries (camd_simple PRIVATE SuiteSparse::CAMD)
 
-  if (TARGET cholmod AND WITH_CHECK)
+  if (WITH_CHOLESKY AND WITH_CHECK)
     add_executable (cholmod_demo
       CHOLMOD/Demo/cholmod_demo.h
       CHOLMOD/Demo/cholmod_demo.c
@@ -1603,7 +1604,7 @@ if (WITH_DEMOS)
 
     add_executable (cholmod_simple CHOLMOD/Demo/cholmod_simple.c)
     target_link_libraries (cholmod_simple PRIVATE SuiteSparse::CHOLMOD)
-  endif (TARGET cholmod AND WITH_CHECK)
+  endif (WITH_CHOLESKY AND WITH_CHECK)
 
   add_executable (colamd_example
     COLAMD/Demo/colamd_example.c
@@ -1719,22 +1720,24 @@ if (WITH_DEMOS)
   endif (WITH_FORTRAN AND CMAKE_Fortran_COMPILER_LOADED)
 
   if (TARGET spqr)
-    add_executable (qrdemo SPQR/Demo/qrdemo.cpp)
-    add_executable (qrdemoc SPQR/Demo/qrdemoc.c)
-    add_executable (qrsimple SPQR/Demo/qrsimple.cpp)
-    add_executable (qrsimplec SPQR/Demo/qrsimplec.c)
+    if (WITH_CHECK)
+      add_executable (qrdemo SPQR/Demo/qrdemo.cpp)
+      add_executable (qrdemoc SPQR/Demo/qrdemoc.c)
+      add_executable (qrsimple SPQR/Demo/qrsimple.cpp)
+      add_executable (qrsimplec SPQR/Demo/qrsimplec.c)
 
-    target_link_libraries (qrdemo PRIVATE SuiteSparse::SPQR)
-    target_link_libraries (qrdemoc PRIVATE SuiteSparse::SPQR)
-    target_link_libraries (qrsimple PRIVATE SuiteSparse::SPQR)
-    target_link_libraries (qrsimplec PRIVATE SuiteSparse::SPQR)
+      target_link_libraries (qrdemo PRIVATE SuiteSparse::SPQR)
+      target_link_libraries (qrdemoc PRIVATE SuiteSparse::SPQR)
+      target_link_libraries (qrsimple PRIVATE SuiteSparse::SPQR)
+      target_link_libraries (qrsimplec PRIVATE SuiteSparse::SPQR)
 
-    list (APPEND SuiteSparse_BIN_TARGETS
-      qrdemo
-      qrdemoc
-      qrsimple
-      qrsimplec
-    )
+      list (APPEND SuiteSparse_BIN_TARGETS
+        qrdemo
+        qrdemoc
+        qrsimple
+        qrsimplec
+      )
+    endif (WITH_CHECK)
 
     if (TARGET gpuqrengine)
       add_executable (qrdemo_gpu SPQR/Demo/qrdemo_gpu.cpp)

